wandb/sdk/wandb_config.py

```diff
@@ -159,17 +210,42 @@ def _update(self, d, allow_val_change=None, ignore_locked=None):
         for key in list(parsed_dict):
             if self._check_locked(key, ignore_locked=ignore_locked):
                 locked_keys.add(key)
-        sanitized = self._sanitize_dict(
+        sanitized, artifacts = self._sanitize_dict(
             parsed_dict, allow_val_change, ignore_keys=locked_keys
         )
         self._items.update(sanitized)
+        self._update_artifact_dict(artifacts)
+        if artifacts:
+            old_artifacts = self._items.get("artifacts", {})
+            sanitized = {**sanitized, "artifacts": {**old_artifacts, **artifacts}}
         return sanitized
 
     def update(self, d, allow_val_change=None):
+        print(d)
+        # don't allow nested dictionary updates of artifacts
```

Comment:
Right now, if a user puts an artifact in the config, its sanitized to `Artifact id_string`. I think limiting nested artifacts makes for an easier implementation now, which we can relax later if need be.